tests/drivers/hwinfo/api/src/main.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2019 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <hwinfo.h>
+#include <ztest.h>
+#include <strings.h>
+
+/*
+ * @addtogroup t_hwinfo_get_device_id_api
+ * @{
+ * @defgroup t_hwinfo_get_device_id test_hwinfo_get_device_id
+ * @brief TestPurpose: verify device id get works
+ * @details
+ * - Test Steps
+ *   -# Read the ID
+ *   -# Check if to many bytes are written to the buffer
+ *   -# Check if UID is plausible
+ * - Expected Results
+ *   -# Device uid with correct length should be written to the buffer.
+ * @}
+ */
+
+#define BUFFER_LENGTH 17
+#define BUFFER_CANARY 0xFF
+
+#if defined(CONFIG_DEVICE_ID_STM32) || defined(CONFIG_DEVICE_ID_NRF)
+
+/*
+ * Function invokes the get_entropy callback in driver
+ * to get the random data and fill to passed buffer
+ */
+static int get_device_id(void)
+{
+	struct device *dev;
+    u8_t buffer_1[BUFFER_LENGTH];
+	u8_t buffer_2[BUFFER_LENGTH];
+	size_t length_read_1, length_read_2;;
+	int i;
+
+	dev = device_get_binding(CONFIG_DEVICE_ID_NAME);
+	if (!dev) {
+		TC_PRINT("error: no device id device\n");
```

Comment:
Ah, and btw, bad news here, I believe TC_PRINT stuff is deprecated, instead we use now zassert_* stuff. @nashif, can you comment?

Reply:
True, according to the latest documentation.
Done